src/ssl.c

```diff
@@ -4807,7 +4807,7 @@ int AddCA(WOLFSSL_CERT_MANAGER* cm, DerBuffer** pDer, int type, int verify)
     }
 #ifndef ALLOW_INVALID_CERTSIGN
     else if (ret == 0 && cert->isCA == 1 && type != WOLFSSL_USER_CA &&
-             (cert->extKeyUsage & KEYUSE_KEY_CERT_SIGN) == 0) {
+        !cert->selfSigned && (cert->extKeyUsage & KEYUSE_KEY_CERT_SIGN) == 0) {
```

Comment:
We have a test case where a root certificate is sent over the wire and it was failing here. Previous code was suppressing this error because the fatal alert logic was processed prior to the call to AddCA, after these changes it showed up. The logic for enforcing keyCertSign applies to intermediate CA's only, so I added the selfSigned check.